### DISCUSSION TITLE
Support for xchacha in reverse mode

### DISCUSSION THREAD

**Discussion by nullmonk** [rfjakob/gocryptfs#759] [Q&A] [answered]
Because reverse mode [hard-codes the cipher to AES-SIV](https://nuetzlich.net/gocryptfs/reverse_mode_crypto/) . Running reverse mode with xchacha produces an error.

```
~ ./gocryptfs --xchacha --deterministic-names --reverse --init files/
Choose a password for protecting your files.
Password: 
Repeat: 
Can't have both XChaCha20Poly1305 and AESSIV feature flags
[exit status 24]
```

Is there anyway to enable xchacha as the cipher while using reverse mode?

**Comment by rfjakob** [comment, marked as answer]
No, unfortunately, this is insecure. Reverse mode needs a chipher that is resistant against nonce reuse, and (x)chacha-poly is not ( https://crypto.stackexchange.com/questions/32075/what-happens-if-a-nonce-is-reused-in-chacha20-poly1305 )
